### PR TITLE
[Core] register flags with name

### PR DESCRIPTION
**Pull request by philbucher**
this is needed if e.g. flags are in a namespace (like the `MapperFlags), then it is no possible to determine the name from the argument

I also cleaned the `IMPLEMENTATION` macro, it is no longer necessary with `NOT` flags being removed

### CODE REVIEW

File: applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp [hunk 1]
```diff
@@ -158,12 +158,12 @@ void ExposeMapperToPython(pybind11::module& m, const std::string& rName)
             ;
 
     // Adding the flags that can be used for mapping
-    mapper.attr("SWAP_SIGN")           = MapperFlags::SWAP_SIGN;
-    mapper.attr("ADD_VALUES")          = MapperFlags::ADD_VALUES;
-    mapper.attr("REMESHED")            = MapperFlags::REMESHED;
-    mapper.attr("USE_TRANSPOSE")       = MapperFlags::USE_TRANSPOSE;
-    mapper.attr("TO_NON_HISTORICAL")   = MapperFlags::TO_NON_HISTORICAL;
-    mapper.attr("FROM_NON_HISTORICAL") = MapperFlags::FROM_NON_HISTORICAL;
+    KRATOS_REGISTER_IN_PYTHON_FLAG_NAME(mapper, MapperFlags::SWAP_SIGN,           "SWAP_SIGN")
```
Comment (adityaghantasala) [inline applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp:161]: why passing `mapper`  again ? you can directly add them to  application module  no ?

Comment (philbucher) [reply applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp:161]: you need to pass the pybind-module to which you want to add the attributes

Comment (philbucher) [reply applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp:161]: Ofc I could also pass the App-module, but I don't want that

**Comment by roigcarlo** [comment]
@philbucher Let's discuss this tomorrow before merge ( It is correct, I just want to make sure, its something we want to add )

File: applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp (hunk 1 above)
Comment (adityaghantasala) [reply applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp:161]: that is what .. curious why  not .. then you do not have to do `MappingApplication.Mapper.FLAG`
but simply `MappingApplication.FLAG`

**Comment by philbucher** [comment]
sure, it is not urgent

**Comment by sunethwarna** [comment]
Does this has a plan to add Flag.Name() method also? If so it would nice, so I can remove flag name passing as string done in HDF5Application. :)

**Pull request closed without merging**